modin/pandas/groupby.py

```diff
@@ -221,19 +221,7 @@ def _shift(data, periods, freq, axis, fill_value, is_set_nan_rows=True):
             return result
 
         if freq is None and axis == 1 and self._axis == 0:
-            # 'by' columns should not be taken into account
```

Comment:
removing workaround for pandas-dev/pandas#42401 bug since it was fixed in 1.3.2.

Now the removed case is handled naturally